openvdb_houdini/houdini/SOP_OpenVDB_Scatter.cc

```diff
@@ -341,96 +379,116 @@ SOP_OpenVDB_Scatter::cookMySop(OP_Context& context)
-                    openvdb::tools::UniformPointScatter<PointAccessor, RandGen, hvdb::Interrupter>
-                        scatter(points, density, mtRand, spread, &boss);
-
-                    if (interior && isSignedDistance) {
-                        processLSInterior(gridType, grid, scatter);
-                    } else {
-                        if (!UTvdbProcessTypedGridTopology(gridType, grid, scatter)) {
-#if UT_MAJOR_VERSION_INT >= 16
-                            UTvdbProcessTypedGridPoint(gridType, grid, scatter);
-#endif
+                    if (ptype == 0) { // vdb points
+
+                        const openvdb::Vec3f dim = openvdb::Vec3f(grid->transform().voxelSize());
+                        const openvdb::Index64 totalPointCount =
+                            openvdb::Index64(density * dim.product()) * grid->activeVoxelCount();
+
+                        using ScatterT = openvdb::points::TotalPointScatter<RandGen, float,
+                                            openvdb::points::NullCodec, openvdb::points::PointDataGrid,
+                                                hvdb::Interrupter>;
+                        ScatterT scatter(totalPointCount, seed, spread, &boss);
+                        if (process(gridType, *grid, scatter, name))  {
+                            pointGrids.push_back(scatter.points());
                         }
                     }
-
-                    if (verbose) scatter.print(gridName);
+                    else { // houdini points
+                        openvdb::tools::UniformPointScatter<PointAccessor, RandGen, hvdb::Interrupter>
+                            scatter(pointAccessor, density, mtRand, spread, &boss);
+                        process(gridType, *grid, scatter, name);
+                    }
                 }
-
             } else if (pmode == 2) { // points per voxel
-
-                openvdb::tools::DenseUniformPointScatter<PointAccessor, RandGen, hvdb::Interrupter>
-                    scatter(points, ptsPerVox, mtRand, spread, &boss);
-
-                if (interior && isSignedDistance) {
-                    processLSInterior(gridType, grid, scatter);
-                } else {
-                    if (!UTvdbProcessTypedGridTopology(gridType, grid, scatter)) {
-#if UT_MAJOR_VERSION_INT >= 16
-                        UTvdbProcessTypedGridPoint(gridType, grid, scatter);
-#endif
+                if (ptype == 0) { // vdb points
+
+                    using ScatterT = openvdb::points::UniformVoxelPointScatter<RandGen, float,
+                        openvdb::points::NullCodec, openvdb::points::PointDataGrid,
+                            hvdb::Interrupter>;
+                    ScatterT scatter(ptsPerVox, seed, spread, &boss);
+                    if (process(gridType, *grid, scatter, name))  {
+                        pointGrids.push_back(scatter.points());
                     }
                 }
-
-                if (verbose) scatter.print(gridName);
```

Comment:
Think you're dropping a few of these verbose statements.

Reply:
They've been moved into process()